src/main/java/ca/carleton/controllers/AccountController.java

```diff
@@ -120,9 +127,16 @@ public ModelAndView requestData(@AuthenticationPrincipal UserDetails userDetails
         if (user == null) {
             model.addAttribute("username", userDetails.getUsername());
             return new ModelAndView("profile-not-found", HttpStatus.NOT_FOUND);
+        } else {
+            if (user.getSubscription()) {
+                return new ModelAndView("redirect:/profile");
+            } else {
+                if (bucket.tryConsume(1))
```

Comment:
Is this per user? Seems like its rate limiting the combined total of all trial users.

Reply:
that is what it does atm ya... Will update it soon